### PR TITLE
azurerm_logic_app_workflow - support for enabled & access_control

### PR DESCRIPTION
**Pull request by neil-yechenwei**
This PR is to add support for new properties [`state`](https://github.com/Azure/azure-rest-api-specs/blob/b9f2042128fbd337c248f57d2b26af6ed2b5bb53/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L6499) and [`accessControl`](https://github.com/Azure/azure-rest-api-specs/blob/b9f2042128fbd337c248f57d2b26af6ed2b5bb53/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L6517) for azurerm_logic_app_workflow.

API failed to create Logic App Workflow with ["accessControl.triggers"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L8956). Per the error message, it complains "OpenAuthenticationAccessPolicy.type" should be "AAD" while creating this resource with "accessControl.triggers", which means ["OpenAuthenticationAccessPolicy.type"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L9019) should be writeable. But after checked swagger, I found this property is marked as [read-only](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L9021). Service team confirmed it's a bug in swagger. So I filed an issue on [rest api spec](https://github.com/Azure/azure-rest-api-specs/issues/16018) and didn't involve it in this PR. Once the bug is fixed, I will submit another PR to add it.

--- PASS: TestAccLogicAppWorkflow_empty (261.74s)
--- PASS: TestAccLogicAppWorkflow_requiresImport (279.18s)
--- PASS: TestAccLogicAppWorkflow_tags (465.55s)
--- PASS: TestAccLogicAppWorkflow_integrationAccount (590.05s)
--- PASS: TestAccLogicAppWorkflow_parameters (240.56s)
--- PASS: TestAccLogicAppWorkflow_accessControl (374.61s)

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Overall this looks good, just one comment that i will leave inline

File: internal/services/logic/logic_app_workflow_resource.go [hunk 1]
```diff
@@ -79,6 +253,16 @@ func resourceLogicAppWorkflow() *pluginsdk.Resource {
 				},
 			},
 
+			"state": {
```
Comment (katbyte) [inline internal/services/logic/logic_app_workflow_resource.go:256]: does this make sense to change to 
```suggestion
			"enabled": {
```
as a boolean to match other resources?

Comment (neil-yechenwei) [reply internal/services/logic/logic_app_workflow_resource.go:256]: Updated

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for your comments. As service team confirmed they allow setting [“OpenAuthenticationAccessPolicy”](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L9015) to only ["accessControl.Triggers"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L8956) for now. But we can set other ["FlowAccessControlConfigurationPolicy"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L8974) props like ["IpAddressRange"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L8978) for ["accessControl.triggers"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L8956), ["accessControl.contents"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L8960), ["accessControl.actions"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L8964) and ["accessControl.workflowManagement"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L8968). So I only implemented ["allowedCallerIpAddresses"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L8978). I intend to implement ["openAuthenticationPolicies"](https://github.com/Azure/azure-rest-api-specs/blob/55d29f1945c8486073b981680da3ddc53d46ce07/specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json#L8986) for "accessControl.triggers" once the [bug](https://github.com/Azure/azure-rest-api-specs/issues/16018) is fixed.

**Review by katbyte — APPROVED** [review]
Thanks @neil-yechenwei - LGTM now 🏗️

**Pull request merged**